src/connection.go

```diff
@@ -44,11 +46,14 @@ func (c configConnectionError) Error() string {
 	return "can't execute redis 'CONFIG' command: " + c.cause.Error()
 }
 
-func newRedisCon(hostname string, port int, unixSocket string, password string) (conn, error) {
-	connectTimeout := redis.DialConnectTimeout(time.Second * 5)
-	readTimeout := redis.DialReadTimeout(time.Second * 5)
-	writeTimeout := redis.DialWriteTimeout(time.Second * 5)
+func newRedisCon(hostname string, port int, unixSocket string, password string, useTLS bool, skipTLSVerify bool) (conn, error) {
```

Comment:
I feel like we have too many arguments in this function. Could it be any other way of modeling this? For example, since we are just passing through some [`DialOption`](https://pkg.go.dev/github.com/garyburd/redigo/redis#DialOption) already perhaps we could create those on `main` 🤔

Reply:
At the beginning I used:
```
func newRedisCon(hostname string, port int, unixSocket string, options ...redis.DialOption) (conn, error) {
```

The issue is that the function in case uses a unixSocket does not need redisTLSTimeout, redisUseTLS and redisSkipTLSVerify. Therefore removing those is annoying receiving a slice

Reply:
we could access directly args, but I would like to avoid accessing global variables in functions called by main

```
func newRedisCon(args argumentList) (conn, error) {

```

Reply:
I am refactoring the whole function